wget www.kernel.org/pub/linux/kernel/vX.Y/linux-X.Y.Z.tar.bz2
tar xvfj linux-X.Y.Z.tar.bz2


wget www.kernel.org/pub/linux/kernel/people/akpm/patches/X.Y/X.Y.Z/X.Y.Z-mm2/X.Y.Z-mm2.bz2

cd /usr/src/linux-X.Y.Z
bzip2 -dc ../X.Y.Z-mm2.bz2 | patch -p1
如果你需要打多个补丁，请注意要采取正确的顺序，为了生成一个包含X.Y.Z-aa-bb补丁的内核，应该首先下载X.Y.Z内核的完整源代码，
再打上X.Y.Z-aa补丁，最后打上X.Y.Z-aa-bb补丁

diff -Nur /path/to/original/kernel /path/to/your/kernel  > changes.patch

/usr 此目录包含了initramfs的实现。

make menuconfig
make
make modules_install
make install

make == make vmlinux && make modules && make bzImage
make zImage
make uImage (mkimage) #z在u-boot中有该命令。

arch/your-arch/configs/your-machine_defconfig


insmod 加载
rmmod  删除
lsmod  查看
modprobe 它先分析/lib/modules/X.Y.Z/modules.dep文件再加载它所依赖的模块
depmod  生成/lib/modules/X.Y.Z/modules.dep文件


make -C /path/to/kernel-soures/ M=`pwd` modules


内存映射表 192.168.10.117                                         内存映射表 192.168.10.71
BIOS-provided physical RAM map:                                   BIOS-provided physical RAM map:
 BIOS-e820: 0000000000000000 - 000000000009d800 (usable)          BIOS-e820: 0000000000000000 - 000000000009d800 (usable)
 BIOS-e820: 000000000009d800 - 00000000000a0000 (reserved)        BIOS-e820: 000000000009d800 - 00000000000a0000 (reserved)
 BIOS-e820: 00000000000e0000 - 0000000000100000 (reserved)        BIOS-e820: 00000000000e0000 - 0000000000100000 (reserved)
 BIOS-e820: 0000000000100000 - 000000007c422000 (usable)          BIOS-e820: 0000000000100000 - 0000000020000000 (usable)
 BIOS-e820: 000000007c422000 - 000000007c530000 (ACPI NVS)        BIOS-e820: 0000000020000000 - 0000000020200000 (reserved)
 BIOS-e820: 000000007c530000 - 000000007f367000 (reserved)        BIOS-e820: 0000000020200000 - 0000000040004000 (usable)
 BIOS-e820: 000000007f367000 - 000000007f3e3000 (ACPI data)       BIOS-e820: 0000000040004000 - 0000000040005000 (reserved)
 BIOS-e820: 000000007f3e3000 - 000000007f4af000 (reserved)        BIOS-e820: 0000000040005000 - 00000000db57c000 (usable)
 BIOS-e820: 000000007f4af000 - 000000007f4b0000 (ACPI NVS)        BIOS-e820: 00000000db57c000 - 00000000db8a4000 (reserved)
 BIOS-e820: 000000007f4b0000 - 000000007f4b9000 (reserved)        BIOS-e820: 00000000db8a4000 - 00000000db8b6000 (ACPI data)
 BIOS-e820: 000000007f4b9000 - 000000007f4c1000 (ACPI NVS)        BIOS-e820: 00000000db8b6000 - 00000000dc28c000 (ACPI NVS)
 BIOS-e820: 000000007f4c1000 - 000000007f4e8000 (reserved)        BIOS-e820: 00000000dc28c000 - 00000000dc84c000 (reserved)
 BIOS-e820: 000000007f4e8000 - 000000007f800000 (ACPI NVS)        BIOS-e820: 00000000dc84c000 - 00000000dc84d000 (usable)
 BIOS-e820: 0000000080000000 - 0000000090000000 (reserved)        BIOS-e820: 00000000dc84d000 - 00000000dc890000 (ACPI NVS)
 BIOS-e820: 00000000fed1c000 - 00000000fed40000 (reserved)        BIOS-e820: 00000000dc890000 - 00000000dcc9d000 (usable)
 BIOS-e820: 00000000ff000000 - 0000000100000000 (reserved)        BIOS-e820: 00000000dcc9d000 - 00000000dcff3000 (reserved)
 BIOS-e820: 0000000100000000 - 0000000c80000000 (usable)          BIOS-e820: 00000000dcff3000 - 00000000dd000000 (usable)
                                                                  BIOS-e820: 00000000dd000000 - 00000000dfa00000 (reserved)
                                                                  BIOS-e820: 00000000f8000000 - 00000000fc000000 (reserved)
                                                                  BIOS-e820: 00000000fec00000 - 00000000fec01000 (reserved)
                                                                  BIOS-e820: 00000000fed00000 - 00000000fed04000 (reserved)
                                                                  BIOS-e820: 00000000fed1c000 - 00000000fed20000 (reserved)
                                                                  BIOS-e820: 00000000fee00000 - 00000000fee01000 (reserved)
                                                                  BIOS-e820: 00000000ff000000 - 0000000100000000 (reserved)
                                                                  BIOS-e820: 0000000100000000 - 000000019f600000 (usable)
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 